src/scripts/main.js

```diff
@@ -1,3 +1,203 @@
+    document.removeEventListener('keyup', movesController);
+  }
+}
+
+function randomNumber() {
+  return Math.floor(Math.random() * gameFieldWidth);
+}
+
+function fillRandomCell() {
+  const [row, col] = [randomNumber(), randomNumber()];
+
+  if (!gameBoard[row][col]) {
+    gameBoard[row][col] = Math.random() > 0.9 ? 4 : 2;
+    handleLose();
+  } else {
+    fillRandomCell();
+  }
+}
+
+function fillBoard() {
+  gameBoard.forEach((row, rowCount) => {
+    row.forEach((_, cellCount) => {
+      const tile = gameField.rows[rowCount].cells[cellCount];
+      const tileValue = gameBoard[rowCount][cellCount];
+
+      tile.classList = 'field-cell';
+      tile.innerText = tileValue;
+
+      if (tileValue > 0) {
+        tile.classList.add('field-cell--' + tileValue);
+      } else {
+        tile.innerText = '';
+      }
+    });
+  });
+}
+
+function duplicateGameBoard() {
+  return gameBoard.map((row) => [...row]);
+}
+
+function compareArrays(newArr, prevArr) {
+  return JSON.stringify(newArr) === JSON.stringify(prevArr);
+}
+
+function flipGridVertical(grid) {
+  return grid.map((row) => [...row].reverse());
+}
+
+function spinGrid(grid) {
+  const newGrid = boardInit();
+
+  newGrid.forEach((row, rowCount) => {
+    row.forEach((_, colCount) => {
+      newGrid[rowCount][colCount] = grid[colCount][rowCount];
+    });
+  });
+
+  return newGrid;
+}
+
+function slideRight(row) {
+  const notEmptyCells = row.filter((cell) => cell > 0);
+  const emptyCells = Array(gameFieldWidth - notEmptyCells.length).fill(0);
+
+  return emptyCells.concat(notEmptyCells);
+}
+
+function mergeCells(arr) {
+  for (let i = gameFieldWidth - 1; i > 0; i--) {
+    if (arr[i] === arr[i - 1]) {
+      arr[i] += arr[i - 1];
+      arr[i - 1] = 0;
+      score += arr[i];
+    }
+  }
+
+  gameScore.innerText = score;
+
+  return arr;
+}
+
+function makeMove(keypress) {
+  let undoFlip = false;
+  let undoSpin = false;
+
+  switch (keypress) {
+    case 'ArrowUp':
+      gameBoard = spinGrid(gameBoard);
+      gameBoard = flipGridVertical(gameBoard);
+      undoFlip = true;
+      undoSpin = true;
+      break;
+    case 'ArrowDown':
+      gameBoard = spinGrid(gameBoard);
+      undoSpin = true;
+      break;
+    case 'ArrowLeft':
+      gameBoard = flipGridVertical(gameBoard);
+      undoFlip = true;
+      break;
+    default:
+      break;
+  }
+
+  const preMergeGrid = duplicateGameBoard();
+
+  gameBoard = gameBoard.map((row) => {
+    return slideRight(mergeCells(slideRight(row)));
+  });
+
+  const preUndoGridsCompare = compareArrays(gameBoard, preMergeGrid);
+
+  if (undoFlip) {
+    gameBoard = flipGridVertical(gameBoard);
+  }
+
+  if (undoSpin) {
+    gameBoard = spinGrid(gameBoard);
+  }
+
+  if (!preUndoGridsCompare) {
+    fillRandomCell();
+  }
+}
+
+function movesController(e) {
+  makeMove(e.key);
+  fillBoard();
+  handleWin();
+}
+
+buttonStart.addEventListener('click', startGame);
+buttonStart.addEventListener('click', restartGame);
```

Comment:
You are adding both 'startGame' and 'restartGame' event listeners to the same 'click' event. Consider refactoring your code to only have one event listener for the 'click' event and to handle starting and restarting the game within that single function.